From 60007d3fe49151dc06a1a92dc655ae141f790c32 Mon Sep 17 00:00:00 2001
From: John Johansen <john.johansen@canonical.com>
Date: Thu, 27 Aug 2020 00:58:55 -0700
Subject: [PATCH] parser: Add warning to capability_table about the need to
 update the Makefile

The autogenerated capabilities comparison needs to be updated if the
capability_table structure changes. Add a note by the structure to
make sure the Makefile will get updated if the structure is changed.

MR: https://gitlab.com/apparmor/apparmor/-/merge_requests/597
Signed-off-by: John Johansen <john.johansen@canonical.com>
Acked-by: Seth Arnold <seth.arnold@canonical.com>
(backported from commit 139fac6062131b11df22bdcb59747329c4f6e7b8)
---
 parser/parser_misc.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/parser/parser_misc.c b/parser/parser_misc.c
index 18e5d2af1..110e01267 100644
--- a/parser/parser_misc.c
+++ b/parser/parser_misc.c
@@ -61,6 +61,11 @@ int is_blacklisted(const char *name, const char *path)
 	return !retval ? 0 : 1;
 }
 
+/*
+ * WARNING: if the format of the following table is changed then
+ *          the Makefile targets, cap_names.h and generated_cap_names.h
+ *          must be updated.
+ */
 struct keyword_table {
 	const char *keyword;
 	unsigned int token;
-- 
2.34.1

